### DISCUSSION TITLE
Question regarding reverse DNS

### DISCUSSION THREAD

**Discussion by sven-borkert** [okd-project/okd#1395] [Q&A] [answered]
Hello all,

I'm trying to understand the requirement of reverse DNS entries for the machines. (https://docs.okd.io/latest/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-network-user-infra_installing-platform-agnostic)

- I'm installing FCOS systems and the hostnames are set via Terraform at installation time. (They don't need the reverse lookup to determine their own hostname)
- The systems automatically get 1 public IP address, I can create a reverse lookup for these.
- I also want to give the systems a private network. Will I need to set up my own DNS server to host the reverse lookups for the 10.x addresses as well, or is it sufficient to have reverse DNS entries for the public IPs?

Regards,
Sven

**Comment by vrutkovs** [comment, marked as answer]
I think documentation is incorrect. If hostnames are provided statically/via DHCP you don't need PTR records.

>Will I need to set up my own DNS server to host the reverse lookups for the 10.x addresses as well, or is it sufficient to have reverse DNS entries for the public IPs?

For some storage drivers nodes need to be able to resolve itself via hostname. You'd want that traffic to be private. Reverse DNS resolution is not required for the cluster to function

**Reply by sven-borkert** [reply]
Thanks for the answer. It seems in the end I needed the PTR records. I did not know at first that I have to install the machines via ignition configs and cannot use pre-installed systems that I have configured the hostname on in advance. Maybe I could have put setting a hostname into the manifest files somehow, but using the PTR records was now the easiest way to get this working.

Regards,
Sven

**Reply by vrutkovs** [reply]
>Maybe I could have put setting a hostname into the manifest files somehow

That's possible, but very inconvenient - that'd make you create separate machineConfigPool for each machine. In that case PTR is a way to go
